source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -43,7 +43,8 @@ Http::Status EnvoyQuicClientStream::encodeHeaders(const Http::RequestHeaderMap&
   SendBufferMonitor::ScopedWatermarkBufferUpdater updater(this, this);
   auto spdy_headers = envoyHeadersToSpdyHeaderBlock(headers);
   if (headers.Method()) {
-    if (headers.Method()->value() == "CONNECT") {
+    if (headers.Method()->value() == "CONNECT" &&
+        !Runtime::runtimeFeatureEnabled("envoy.reloadable_features.use_rfc_connect")) {
```

Comment:
I'm slightly surprised not to see an `else` for the case where the runtime flag is enabled to be parallel with the logic in the h2 code. Is that not required?